### PR TITLE
Updated links to Assertion in Managed Code docs

### PR DESCRIPTION
**Pull request by jmlane**
# Updated Exceptions documentation links to latest Assertion in Managed Code documentation

## Summary

Old links for [Assertions in Managed Code](http://msdn.microsoft.com/library/70ab2522-6486-4076-a1a9-e0f11cd0f3a1) were pointing to the MSDN page. I updated these, pointing to the latest docs.microsoft.com article for [Assertions in Managed Code].

[Assertions in Managed Code]: https://docs.microsoft.com/en-ca/visualstudio/debugger/assertions-in-managed-code

### CODE REVIEW

File: xml/System/Exception.xml [hunk 1]
```diff
@@ -78,7 +78,7 @@
...
      [!code-vb[System.Exception.Class#5](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.exception.class/vb/usageerrors2.vb#5)]  
   
-     Instead of using exception handling for usage errors, you can use the <xref:System.Diagnostics.Debug.Assert%2A?displayProperty=nameWithType> method to identify usage errors in debug builds, and the <xref:System.Diagnostics.Trace.Assert%2A?displayProperty=nameWithType> method to identify usage errors in both debug and release builds. For more information, see [Assertions in Managed Code](http://msdn.microsoft.com/library/70ab2522-6486-4076-a1a9-e0f11cd0f3a1).  
+     Instead of using exception handling for usage errors, you can use the <xref:System.Diagnostics.Debug.Assert%2A?displayProperty=nameWithType> method to identify usage errors in debug builds, and the <xref:System.Diagnostics.Trace.Assert%2A?displayProperty=nameWithType> method to identify usage errors in both debug and release builds. For more information, see [Assertions in Managed Code](https://docs.microsoft.com/en-ca/visualstudio/debugger/assertions-in-managed-code).
```
Comment (mairaw) [inline xml/System/Exception.xml]: change the link from https://docs.microsoft.com/en-ca/visualstudio/debugger/assertions-in-managed-code to just /visualstudio/debugger/assertions-in-managed-code

File: xml/System/Exception.xml [hunk 2]
```diff
@@ -122,7 +122,7 @@
...
+ Throwing or handling an exception consumes a significant amount of system resources and execution time. Throw exceptions only to handle truly extraordinary conditions, not to handle predictable events or flow control. For example, in some cases, such as when you're developing a class library, it's reasonable to throw an exception if a method argument is invalid, because you expect your method to be called with valid parameters. An invalid method argument, if it is not the result of a usage error, means that something extraordinary has occurred. Conversely, do not throw an exception if user input is invalid, because you can expect users to occasionally enter invalid data. Instead, provide a retry mechanism so users can enter valid input. Nor should you use exceptions to handle usage errors. Instead, use [assertions](https://docs.microsoft.com/en-ca/visualstudio/debugger/assertions-in-managed-code) to identify and correct usage errors.
```
Comment (mairaw) [inline xml/System/Exception.xml]: same here

**Review by mairaw — CHANGES_REQUESTED** [review]
Thanks for replacing these links @jmlane. I just left some feedback on how we usually do those so they work online across all locales and in our offline books too.

**Comment by mairaw** [comment]
Related to #2037

**Review by mairaw — APPROVED** [review]

**Comment by mairaw** [comment]
Thanks @jmlane. I've applied the feedback I've left you and also made the same changes to two other topics that had the same link. Once build finishes, we'll merge this PR and the fix should be live tomorrow or so.

**Comment by mairaw** [comment]
I actually removed the two additional files because it was requiring us to have a CLA. I'll create a separate PR for the two other files.

**Comment by jmlane** [comment]
Thanks for following up @mairaw. I signed the CLA. If you want to use this or close it in favour of another PR, go right ahead.

**Comment by mairaw** [comment]
Thanks @jmlane. I've moved the additional changes to #4080 and kept only the original changes here.

As soon as build finishes, I'll merge your PR. Thanks for your contributions to our docs!

**Pull request merged**